tests/scripts/all.sh

```diff
@@ -1572,36 +1576,36 @@ component_test_no_64bit_multiplication () {
 }
 
 component_build_arm_none_eabi_gcc () {
-    msg "build: arm-none-eabi-gcc, make" # ~ 10s
+    msg "build: ${ARM_GCC_PREFIX}gcc, make" # ~ 10s
```

Comment:
I'm not sure, and I'd prefer to defer such a change to when we know how we'll use this feature. _If_ we'll use this feature.

I initially did this because I thought it would be necessary to use a different compiler on Travis to avoid having to install `arm-none-eabi-gcc`. But it turned out that the compiler we're using on Jenkins is from Ubuntu, so it was straightforward to add this compiler. I left the commit in because even if it isn't necessary now, it could be useful later.

There are two ways this can be used: to substitute a different platform, or to use a compiler installed at a location outside `$PATH`. For the first use case, we should have different component names. For the second use case, the component should keep its name. My current preference is to declare that the second use case is the one that's primarily intended, and so keep the component names.

Reply:
Well, I am not really convinced. If ${ARM_GCC_PREFIX} is set up to a value that has nothing to do with arm_none_eabi_gcc then the component has nothing to do with arm_none_eabi_gcc but is called component_build_arm_none_eabi_gcc () which is confusing to me.
Regarding the second use case, making the path to arm_none_eabi_gcc configurable would be enough it seems.
I understand that we are not completely committed to the second use case thus the idea to let it not very well defined  yet. But we then go from a situation where things are clear to a situation where things are unclear: the component is named component_build_arm_none_eabi_gcc () but it may not use at all arm_none_eabi_gcc compiler.

Reply:
How about renaming `ARM_GCC_PREFIX` to `ARM_NONE_EABI_GCC_PREFIX`?

Reply:
Yes, it would be fine by me with this renaming.